engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterLaunchEngine.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_DARWIN_IOS_FRAMEWORK_SOURCE_FLUTTERLAUNCHENGINE_H_
+#define FLUTTER_SHELL_PLATFORM_DARWIN_IOS_FRAMEWORK_SOURCE_FLUTTERLAUNCHENGINE_H_
+
+#import "flutter/shell/platform/darwin/ios/framework/Headers/FlutterEngine.h"
+
+/**
+ * A lazy container for an engine that will only dispense one engine.
+ *
+ * This is used to hold an engine for plugin registration when the
+ * GeneratedPluginRegistrant is called on a FlutterAppDelegate before the first
+ * FlutterViewController is set up. This is the typical flow after the
+ * UISceneDelegate migration.
+ */
+@interface FlutterLaunchEngine : NSObject
+
+/**
+ * Accessor for the launch engine.
+ */
+@property(nonatomic, strong, nullable, readonly) FlutterEngine* engine;
+
+/**
+ * Grab ownership of the launch engine.
+ *
+ * After this is called `self.engine` and `grabEngine` will always return nil.
+ */
+- (nullable FlutterEngine*)grabEngine;
```

Comment:
it's hard to understand what "grab" means when reading the code. But I couldn't think of a better name. Maybe `moveOwnership`?

Reply:
I renamed it "take*".  "Move" has different semantics in objective-c++.

Reply:
Makes sense. Move feels like C++ move (tho to some extend this is similar concept as it invalidates the `FlutterLaunchEngine`)